static/js/narrow.js

```diff
@@ -295,6 +295,10 @@ exports.activate = function (raw_operators, opts) {
     typing_events.render_notifications_for_narrow();
     tab_bar.initialize();
 
+    if (activity.is_initialized()) {
+        activity.redraw();
+    }
```

Comment:
Do we need to do a redraw unconditionally here, or can we check whether this setting is enabled?  Redraws are really expensive in the `all` state...

(Also, do we need to check for `activity.is_initialized`?  Like, is there an actual code path where it wouldn't be?).

Reply:
You're right,
In fact, maybe I should maintain a duplicate list just to ensure that we actually do need a redraw...

will look into it. 
(and yeah, maybe that check on `activity.is_initialized()` is unnecessary...)